# Data Science Homework-3
#### Name: Muhamamd Talha Salani
#### ID: ms07725

Purpose:
Demonstrate exploration of data via Hypothesis Testing methods using Python and tell
interesting stories and insights around your statistical inferences.
Points: 100
Perform Hypothesis Testing to choose between Null and Alternative Hypothesis and provide
justification to your decision with explanation.
1. If math_score of males is greater than females.
2. If math_score of Non Doctors is greater than Doctors (Hint: Use career_aspirations to
find Doctor and Non Doctors)
3. If weekly_self_study_hours of males are greater than females.
4. If math_score of no part_time_job male students are greater than part_
time_job male
students.
5. If the average of history_score is less than geography_score.
6. If the average of math_score is greater than biology_score.
7. If weekly_self_study_hours of extracurricular_activities is less than no
extracurricular_activities.
8. If absence_days of females are less than males.
9. Is there any Association between gender and career_aspiration.
10. Is there any Association between gender and part_time_job.

For each question, write Null and Alternative Hypothesis first. Then interpret all the results
with your final decision on Hypothesis Testing. No points will be given without explanation.

Deliverable: Submit a ipynb file containing your code, outputs and explanations. Include
homework title, your name and your email on top of your ipynb code file.

In [1044]:
import pandas as pd
import numpy as np
from scipy import stats

In [1045]:
df = pd.read_csv('student-scores.csv')

In [1046]:
df.isnull().sum()

id                            0
first_name                    0
last_name                     0
email                         0
gender                        0
part_time_job                 0
absence_days                  0
extracurricular_activities    0
weekly_self_study_hours       0
career_aspiration             0
math_score                    0
history_score                 0
physics_score                 0
chemistry_score               0
biology_score                 0
english_score                 0
geography_score               0
dtype: int64

In [1047]:
df.head()

id first_name last_name                                  email  gender  \
0   1       Paul     Casey         paul.casey.1@gslingacademy.com    male   
1   2   Danielle  Sandoval  danielle.sandoval.2@gslingacademy.com  female   
2   3       Tina   Andrews       tina.andrews.3@gslingacademy.com  female   
3   4       Tara     Clark         tara.clark.4@gslingacademy.com  female   
4   5    Anthony    Campos     anthony.campos.5@gslingacademy.com    male   

   part_time_job  absence_days  extracurricular_activities  \
0          False             3                       False   
1          False             2                       False   
2          False             9                        True   
3          False             5                       False   
4          False             5                       False   

   weekly_self_study_hours   career_aspiration  math_score  history_score  \
0                       27              Lawyer          73             81   
1                       47              Doctor          90             86   
2                       13  Government Officer          81             97   
3                        3              Artist          71             74   
4                       10             Unknown          84             77   

   physics_score  chemistry_score  biology_score  english_score  \
0             93               97             63             80   
1             96              100             90             88   
2             95               96             65             77   
3             88               80             89             63   
4             65               65             80             74   

   geography_score  
0               87  
1               90  
2               94  
3               86  
4               76

### Q1: If math_score of males is greater than females.

#### Setting up the Null, and Alternative hypothesis:

Null Hypothesis $H_{o}$ : math_score of males and females are same.

Alternative Hypothesis $H_{alt}$ : math_score of males is greater than the math_score of females.

In [1050]:
males = df[df['gender']=='male']['math_score']
females = df[df['gender']=='female']['math_score']


t_stat, p_value = stats.ttest_ind(males, females, alternative='greater')
print("t-statistic:", t_stat, "p-value:", p_value)

t-statistic: 2.2371559026936256 p-value: 0.01269302029963023


#### Interpretation of Results
- **t-statistic**: A positive t-statistic (2.24) indicates that the sample mean math score of males is higher than that of females.
- **p-value**: A p-value of 0.0127 is less than the significance level of 0.05, meaning that the observed difference in math scores between males and females is statistically significant.

#### Conclusion
Since the **p-value is less than 0.05**, we **reject the null hypothesis**. This suggests that **males have significantly higher math scores than females** based on the data.

### Q2: If math_score of Non Doctors is greater than Doctors (Hint: Use career_aspirations to find Doctor and Non Doctors)

In [1053]:
df.head()

id first_name last_name                                  email  gender  \
0   1       Paul     Casey         paul.casey.1@gslingacademy.com    male   
1   2   Danielle  Sandoval  danielle.sandoval.2@gslingacademy.com  female   
2   3       Tina   Andrews       tina.andrews.3@gslingacademy.com  female   
3   4       Tara     Clark         tara.clark.4@gslingacademy.com  female   
4   5    Anthony    Campos     anthony.campos.5@gslingacademy.com    male   

   part_time_job  absence_days  extracurricular_activities  \
0          False             3                       False   
1          False             2                       False   
2          False             9                        True   
3          False             5                       False   
4          False             5                       False   

   weekly_self_study_hours   career_aspiration  math_score  history_score  \
0                       27              Lawyer          73             81   
1                       47              Doctor          90             86   
2                       13  Government Officer          81             97   
3                        3              Artist          71             74   
4                       10             Unknown          84             77   

   physics_score  chemistry_score  biology_score  english_score  \
0             93               97             63             80   
1             96              100             90             88   
2             95               96             65             77   
3             88               80             89             63   
4             65               65             80             74   

   geography_score  
0               87  
1               90  
2               94  
3               86  
4               76

#### Setting up the Null, and Alternative hypothesis:

Null Hypothesis $H_{o}$ : The mean math scores of non-doctors are less than or equal to the mean math scores of doctors.

Alternative Hypothesis $H_{alt}$ : The mean math scores of non-doctors are greater than those of doctors.

In [1055]:
non_doctors = df[df['career_aspiration']!='Doctor']['math_score']
doctors = df[df['career_aspiration']=='Doctor']['math_score']

t_stat, p_value = stats.ttest_ind(non_doctors, doctors, alternative='greater')
print("t-statistic:", t_stat, "p-value:", p_value)


t-statistic: -7.389083205876682 p-value: 0.999999999999892


#### Interpretation of Results
- **t-statistic**: A negative t-statistic (-7.39) indicates that the sample mean math score of non-doctors is lower than the sample mean math score of doctors. 
- **p-value**: A p-value of approximately 1 means that the difference between the two groups is not statistically significant, and we cannot reject the null hypothesis.

#### Conclusion
Since the **p-value is much greater than 0.05**, we **fail to reject the null hypothesis**. This suggests that **there is no evidence to conclude that non-doctors have higher math scores than doctors**. In fact, the negative t-statistic suggests the opposite—that doctors may have higher math scores, though the difference is not statistically significant.

### Q3 If weekly_self_study_hours of males are greater than females.

In [1058]:
df.head()

id first_name last_name                                  email  gender  \
0   1       Paul     Casey         paul.casey.1@gslingacademy.com    male   
1   2   Danielle  Sandoval  danielle.sandoval.2@gslingacademy.com  female   
2   3       Tina   Andrews       tina.andrews.3@gslingacademy.com  female   
3   4       Tara     Clark         tara.clark.4@gslingacademy.com  female   
4   5    Anthony    Campos     anthony.campos.5@gslingacademy.com    male   

   part_time_job  absence_days  extracurricular_activities  \
0          False             3                       False   
1          False             2                       False   
2          False             9                        True   
3          False             5                       False   
4          False             5                       False   

   weekly_self_study_hours   career_aspiration  math_score  history_score  \
0                       27              Lawyer          73             81   
1                       47              Doctor          90             86   
2                       13  Government Officer          81             97   
3                        3              Artist          71             74   
4                       10             Unknown          84             77   

   physics_score  chemistry_score  biology_score  english_score  \
0             93               97             63             80   
1             96              100             90             88   
2             95               96             65             77   
3             88               80             89             63   
4             65               65             80             74   

   geography_score  
0               87  
1               90  
2               94  
3               86  
4               76

#### Setting up the Null, and Alternative hypothesis:

Null Hypothesis $H_{o}$ : The mean weekly self-study hours of males are equal to or less than those of females.

Alternative Hypothesis $H_{alt}$ : The mean weekly self-study hours of males are greater than those of females.

In [1060]:
male_study_hours = df[df['gender'] == 'male']['weekly_self_study_hours']
female_study_hours = df[df['gender'] == 'female']['weekly_self_study_hours']


t_stat, p_value = stats.ttest_ind(male_study_hours, female_study_hours, alternative='greater')
print("t-statistic:", t_stat, "p-value:", p_value)

t-statistic: 3.4567831650474443 p-value: 0.0002790501841844182


#### Interpretation of Results

- **t-statistic**: A positive t-statistic (3.46) indicates that the sample mean weekly self-study hours of males is higher than that of females.
- **p-value**: A p-value of 0.00028 is much smaller than the significance level of 0.05, meaning that the observed difference in weekly self-study hours between males and females is statistically significant.

#### Conclusion
Since the p-value is much less than 0.05, we reject the null hypothesis. This suggests that males spend significantly more time on self-study than females based on the data.

### Q4 If math_score of no part_time_job male students are greater than part_time_job male students.

In [1063]:
df.head()

id first_name last_name                                  email  gender  \
0   1       Paul     Casey         paul.casey.1@gslingacademy.com    male   
1   2   Danielle  Sandoval  danielle.sandoval.2@gslingacademy.com  female   
2   3       Tina   Andrews       tina.andrews.3@gslingacademy.com  female   
3   4       Tara     Clark         tara.clark.4@gslingacademy.com  female   
4   5    Anthony    Campos     anthony.campos.5@gslingacademy.com    male   

   part_time_job  absence_days  extracurricular_activities  \
0          False             3                       False   
1          False             2                       False   
2          False             9                        True   
3          False             5                       False   
4          False             5                       False   

   weekly_self_study_hours   career_aspiration  math_score  history_score  \
0                       27              Lawyer          73             81   
1                       47              Doctor          90             86   
2                       13  Government Officer          81             97   
3                        3              Artist          71             74   
4                       10             Unknown          84             77   

   physics_score  chemistry_score  biology_score  english_score  \
0             93               97             63             80   
1             96              100             90             88   
2             95               96             65             77   
3             88               80             89             63   
4             65               65             80             74   

   geography_score  
0               87  
1               90  
2               94  
3               86  
4               76

Null Hypothesis $H_{o}$ : The mean math scores of males with no part-time job are less than or equal to those of males with part-time jobs.

Alternative Hypothesis $H_{alt}$ : The mean math scores of males with no part-time job are greater than those of males with part-time jobs.

In [1065]:
no_job_males = df[(df['gender'] == 'male') & (df['part_time_job'] == False)]['math_score']
job_males = df[(df['gender'] == 'male') & (df['part_time_job'] == True)]['math_score']

t_stat, p_value = stats.ttest_ind(no_job_males, job_males, alternative='greater')
print("t-statistic:", t_stat, "p-value:", p_value)

t-statistic: 5.8026672393780085 p-value: 4.3820239750124745e-09


#### Interpretation of Results

- **t-statistic**: A high positive t-statistic (5.80) suggests that male students without part-time jobs have significantly higher math scores than those with part-time jobs.
- **p-value**: A p-value of 4.38e-09 is extremely small and far below the significance level of 0.05. This means the observed difference is statistically significant.

#### Conclusion
Since the p-value is much less than 0.05, we reject the null hypothesis. This indicates that male students without part-time jobs have significantly higher math scores than those with part-time jobs based on the data.

### Q5 If the average of history_score is less than geography_score.

In [1068]:
df.head()

id first_name last_name                                  email  gender  \
0   1       Paul     Casey         paul.casey.1@gslingacademy.com    male   
1   2   Danielle  Sandoval  danielle.sandoval.2@gslingacademy.com  female   
2   3       Tina   Andrews       tina.andrews.3@gslingacademy.com  female   
3   4       Tara     Clark         tara.clark.4@gslingacademy.com  female   
4   5    Anthony    Campos     anthony.campos.5@gslingacademy.com    male   

   part_time_job  absence_days  extracurricular_activities  \
0          False             3                       False   
1          False             2                       False   
2          False             9                        True   
3          False             5                       False   
4          False             5                       False   

   weekly_self_study_hours   career_aspiration  math_score  history_score  \
0                       27              Lawyer          73             81   
1                       47              Doctor          90             86   
2                       13  Government Officer          81             97   
3                        3              Artist          71             74   
4                       10             Unknown          84             77   

   physics_score  chemistry_score  biology_score  english_score  \
0             93               97             63             80   
1             96              100             90             88   
2             95               96             65             77   
3             88               80             89             63   
4             65               65             80             74   

   geography_score  
0               87  
1               90  
2               94  
3               86  
4               76

Null Hypothesis $H_{o}$ : The mean history score is greater than or equal to the mean geography score.

Alternative Hypothesis $H_{alt}$ : The mean history score is less than the mean geography score.

In [1070]:
t_stat, p_value = stats.ttest_rel(df['history_score'], df['geography_score'], alternative='less')
print("t-statistic:", t_stat, "p-value:", p_value)

t-statistic: -1.4909007828207366 p-value: 0.06807265919151075


#### Interpretation of Results

- **t-statistic**: A negative t-statistic (-1.49) suggests that the sample mean history score is lower than the sample mean geography score.
- **p-value**: A p-value of 0.0681 is slightly above the typical significance level of 0.05, meaning the observed difference is not statistically significant at this threshold.

#### Conclusion
Since the p-value is greater than 0.05, we fail to reject the null hypothesis. This suggests that there is insufficient evidence to conclude that the average history score is less than the average geography score based on the data. However, the t-statistic does suggest history scores might be lower, though not at a statistically significant level.

### Q6 If the average of math_score is greater than biology_score.

In [1073]:
df.head()

id first_name last_name                                  email  gender  \
0   1       Paul     Casey         paul.casey.1@gslingacademy.com    male   
1   2   Danielle  Sandoval  danielle.sandoval.2@gslingacademy.com  female   
2   3       Tina   Andrews       tina.andrews.3@gslingacademy.com  female   
3   4       Tara     Clark         tara.clark.4@gslingacademy.com  female   
4   5    Anthony    Campos     anthony.campos.5@gslingacademy.com    male   

   part_time_job  absence_days  extracurricular_activities  \
0          False             3                       False   
1          False             2                       False   
2          False             9                        True   
3          False             5                       False   
4          False             5                       False   

   weekly_self_study_hours   career_aspiration  math_score  history_score  \
0                       27              Lawyer          73             81   
1                       47              Doctor          90             86   
2                       13  Government Officer          81             97   
3                        3              Artist          71             74   
4                       10             Unknown          84             77   

   physics_score  chemistry_score  biology_score  english_score  \
0             93               97             63             80   
1             96              100             90             88   
2             95               96             65             77   
3             88               80             89             63   
4             65               65             80             74   

   geography_score  
0               87  
1               90  
2               94  
3               86  
4               76

Null Hypothesis $H_{o}$ : The mean math score is less than or equal to the mean biology score.

Alternative Hypothesis $H_{alt}$ : The mean math score is greater than the mean biology score.

In [1075]:
t_stat, p_value = stats.ttest_rel(df['math_score'], df['biology_score'], alternative='greater')
print("t-statistic:", t_stat, "p-value:", p_value)

t-statistic: 9.475687955847599 p-value: 3.6050784652882275e-21


#### Interpretation of Results
- **t-statistic**: A highly positive t-statistic (9.48) suggests that the sample mean math score is much higher than the sample mean biology score.
- **p-value**: A p-value of 3.61e-21 is extremely small and far below the significance level of 0.05. This means that the observed difference in math and biology scores is statistically significant.

#### Conclusion
Since the p-value is far less than 0.05, we reject the null hypothesis. This suggests that the average math score is significantly higher than the average biology score based on the data.

### Q7 If weekly_self_study_hours of extracurricular_activities is less than no extracurricular_activities.

In [1078]:
df.head()

id first_name last_name                                  email  gender  \
0   1       Paul     Casey         paul.casey.1@gslingacademy.com    male   
1   2   Danielle  Sandoval  danielle.sandoval.2@gslingacademy.com  female   
2   3       Tina   Andrews       tina.andrews.3@gslingacademy.com  female   
3   4       Tara     Clark         tara.clark.4@gslingacademy.com  female   
4   5    Anthony    Campos     anthony.campos.5@gslingacademy.com    male   

   part_time_job  absence_days  extracurricular_activities  \
0          False             3                       False   
1          False             2                       False   
2          False             9                        True   
3          False             5                       False   
4          False             5                       False   

   weekly_self_study_hours   career_aspiration  math_score  history_score  \
0                       27              Lawyer          73             81   
1                       47              Doctor          90             86   
2                       13  Government Officer          81             97   
3                        3              Artist          71             74   
4                       10             Unknown          84             77   

   physics_score  chemistry_score  biology_score  english_score  \
0             93               97             63             80   
1             96              100             90             88   
2             95               96             65             77   
3             88               80             89             63   
4             65               65             80             74   

   geography_score  
0               87  
1               90  
2               94  
3               86  
4               76

Null Hypothesis $H_{o}$ : The mean weekly self-study hours of students with extracurricular activities are greater than or equal to those with no extracurricular activities.

Alternative Hypothesis $H_{alt}$ : The mean weekly self-study hours of students with extracurricular activities are less than those with no extracurricular activities.

In [1080]:
with_activities = df[df['extracurricular_activities'] == True]['weekly_self_study_hours']
no_activities = df[df['extracurricular_activities'] == False]['weekly_self_study_hours']

t_stat, p_value = stats.ttest_ind(with_activities, no_activities, alternative='less')
print("t-statistic:", t_stat, "p-value:", p_value)

t-statistic: -2.3506159912870577 p-value: 0.009419553161020007


#### Interpretation of Results
- **t-statistic**: A negative t-statistic (-2.35) suggests that students involved in extracurricular activities tend to spend less time on self-study compared to those not involved in extracurricular activities.
- **p-value**: A p-value of 0.0094 is less than the significance level of 0.05, indicating that the difference in weekly self-study hours between the two groups is statistically significant.

#### Conclusion
Since the p-value is less than 0.05, we reject the null hypothesis. This suggests that students involved in extracurricular activities spend significantly less time on weekly self-study compared to those who are not involved in extracurricular activities based on the data.

### Q8 If absence_days of females are less than males.

In [1083]:
df.head()

id first_name last_name                                  email  gender  \
0   1       Paul     Casey         paul.casey.1@gslingacademy.com    male   
1   2   Danielle  Sandoval  danielle.sandoval.2@gslingacademy.com  female   
2   3       Tina   Andrews       tina.andrews.3@gslingacademy.com  female   
3   4       Tara     Clark         tara.clark.4@gslingacademy.com  female   
4   5    Anthony    Campos     anthony.campos.5@gslingacademy.com    male   

   part_time_job  absence_days  extracurricular_activities  \
0          False             3                       False   
1          False             2                       False   
2          False             9                        True   
3          False             5                       False   
4          False             5                       False   

   weekly_self_study_hours   career_aspiration  math_score  history_score  \
0                       27              Lawyer          73             81   
1                       47              Doctor          90             86   
2                       13  Government Officer          81             97   
3                        3              Artist          71             74   
4                       10             Unknown          84             77   

   physics_score  chemistry_score  biology_score  english_score  \
0             93               97             63             80   
1             96              100             90             88   
2             95               96             65             77   
3             88               80             89             63   
4             65               65             80             74   

   geography_score  
0               87  
1               90  
2               94  
3               86  
4               76

Null Hypothesis $H_{o}$ : The mean absence days of females are greater than or equal to those of males.

Alternative Hypothesis $H_{alt}$ : The mean absence days of females are less than those of males.

In [1085]:
female_absence = df[df['gender'] == 'female']['absence_days']
male_absence = df[df['gender'] == 'male']['absence_days']

t_stat, p_value = stats.ttest_ind(female_absence, male_absence, alternative='less')
print("t-statistic:", t_stat, "p-value:", p_value)


t-statistic: 1.3128102331149203 p-value: 0.9053011661354035


#### Interpretation of Results

- **t-statistic**: A positive t-statistic (1.31) suggests that the sample mean absence days of females is slightly higher than that of males, but not by a large margin.
- **p-value**: A p-value of 0.9053 is much larger than the significance level of 0.05, meaning that the observed difference in absence days between females and males is not statistically significant.

#### Conclusion
Since the p-value is much greater than 0.05, we fail to reject the null hypothesis. This suggests that there is no significant evidence to conclude that the absence days of females are less than those of males based on the data. In fact, the result indicates no significant difference in absence days between genders.








### Q9 Is there any Association between gender and career_aspiration.

Null Hypothesis $H_{o}$ : There is no association between gender and career_aspiration.

Alternative Hypothesis $H_{alt}$ : There is an association between gender and career_aspiration.

In [1089]:
contingency_table = pd.crosstab(df['gender'], df['career_aspiration'])
chi2, p, dof, expected = stats.chi2_contingency(contingency_table)
print("Chi-square:", chi2, "p-value:", p)


Chi-square: 155.39466648000172 p-value: 6.718966009926925e-25


#### Interpretation of Results

Chi-square statistic: A large Chi-square value (155.39) indicates a significant deviation from the assumption that gender and career aspiration are independent.
p-value: The p-value is extremely small (6.72e-25), far below the significance level of 0.05. This means that the observed association between gender and career aspiration is statistically significant.

#### Conclusion
Since the p-value is far less than 0.05, we reject the null hypothesis. This suggests that there is a significant association between gender and career aspiration based on the data. Gender and career aspirations are not independent of each other.

### Q10 Is there any Association between gender and part_time_job.

Null Hypothesis $H_{o}$ : There is no association between gender and part-time job.

Alternative Hypothesis $H_{alt}$ : There is an association between gender and part-time job.

In [1093]:
contingency_table = pd.crosstab(df['gender'], df['part_time_job'])
chi2, p, dof, expected = stats.chi2_contingency(contingency_table)
print("Chi-square:", chi2, "p-value:", p)


Chi-square: 1.0069172842312688 p-value: 0.31564249665103933


#### Interpretation of Results

Chi-square statistic: A relatively low Chi-square value (1.01) suggests little deviation from the assumption that gender and part-time job status are independent.
p-value: A p-value of 0.3156 is greater than the significance level of 0.05, indicating that the observed association between gender and part-time job status is not statistically significant.

#### Conclusion
Since the p-value is greater than 0.05, we fail to reject the null hypothesis. This suggests that there is no significant association between gender and part-time job status based on the data. Gender and part-time job status appear to be independent.